source/Ship.cpp

```diff
@@ -1300,6 +1300,31 @@ void Ship::Place(Point position, Point velocity, Angle angle, bool isDeparting)
 				bay.ship->SetSwizzle(bay.ship->customSwizzle >= 0 ? bay.ship->customSwizzle : swizzle);
 		}
 	}
+		// Get the weapon ranges for this ship, so the Ai can call it.
+	const vector<Hardpoint> &hardpoints = armament.Get();
+	for(unsigned i = 0; i < hardpoints.size(); ++i)
+	{
+		const Weapon *weapon = hardpoints[i].GetOutfit();
+		if(weapon && !weapon->Ammo() && weapon->DoesDamage() && hardpoints[i].IsTurret())
+			turretRange = max(turretRange, weapon->Range() + hardpoints[i].GetPoint().Length());
+		else if(weapon && !weapon->Ammo() && weapon->DoesDamage() && hardpoints[i].IsGun())
+			gunRange = max(gunRange, weapon->Range() + hardpoints[i].GetPoint().Length());
+	}
```

Comment:
this is c style, lets upgrade this
```suggestion
	for(const auto &hardpoint : armament.Get())
	{
		const Weapon *weapon = hardpoint.GetOutfit();
		if(weapon && !weapon->Ammo() && weapon->DoesDamage() && hardpoint.IsTurret())
			turretRange = max(turretRange, weapon->Range() + hardpoint.GetPoint().Length());
		else if(weapon && !weapon->Ammo() && weapon->DoesDamage() && hardpoint.IsGun())
			gunRange = max(gunRange, weapon->Range() + hardpoint.GetPoint().Length());
	}
```